pkg/scale/encode.go

```diff
@@ -223,6 +226,40 @@ func (es *encodeState) encodeArray(in interface{}) (err error) {
 	return
 }
 
+func (es *encodeState) encodeMap(in interface{}) (err error) {
+	v := reflect.ValueOf(in)
+	err = es.encodeLength(v.Len())
+	if err != nil {
+		return fmt.Errorf("encoding length: %w", err)
+	}
+
+	mapKeys := v.MapKeys()
+
+	sort.Slice(mapKeys, func(i, j int) bool {
+		keyByteOfI, _ := Marshal(mapKeys[i].Interface())
+		keyByteOfJ, _ := Marshal(mapKeys[j].Interface())
+		return bytes.Compare(keyByteOfI, keyByteOfJ) < 0
+	})
```

Comment:
yeah

Reply:
@axaysagathiya can you create another PR to only do this in the tests? This is not required when encoding maps in production, and it's not very performant.  I'd suggest adding a `sortKeys` parameter to the `encodeMap` function signature.

Reply:
Oh, my bad! 
> This is not required when encoding maps

This is not just needed in test, this is needed in encoding as well. keys in go maps are unordered. so running encode function on the same go map could result into different encodings. Rust equivalent of go map is BTree map, where keys are sorted. Scale encode function in the Rust implementation does not produces different values on running it multiple times.

So, in order to make sure that our scale package and Rust's scale package encodes a map to the same value sorting is necessary.

Reply:
Just because the `BTreeMap` has deterministic sorting of keys, doesn't mean that the `parity-scale-codec` isn't able to decode maps that were encoded with keys in non-deterministic order.  In go it's understood that the order of the keys is non-deterministic.  In this case, we should be testing that we can encode a map and decode into a map, and assert that the maps are equal.  We could also ensure that decoding into a rust `BTreeMap` via the `parity-scale-codec` works as expected.  To make the tests more deterministic, you could provide a map with a single key or a map with two keys, and just assert that it's equal to one of the two encodings and remove the sorting logic altogether.